src/commands/new.ts

```diff
@@ -12,84 +12,291 @@ import {
+   */
+  coloLoco?: boolean
+  /**
+   * Log raw parameters for debugging, run formatting script not quietly
+   *
+   * Input Source: `parameter.option`
+   * @default false
+   */
+  debug?: boolean
+  /**
+   * Create new git repository and create an inital commit with boilerplate changes
+   *
+   * Input Source: `prompt.confirm` | `parameter.option`
+   * @default true
+   */
+  git?: boolean
+  /**
+   * Whether or not to run packager install script after project is created
+   *
+   * Input Source: `prompt.confirm` | `parameter.option`
+   * @default true
+   */
+  installDeps?: boolean
+  /**
+   * Remove existing directory otherwise throw if exists
+   *
+   * Input Source: `prompt.confirm` | `parameter.option`
+   * @default false
+   */
+  overwrite?: boolean
+  /**
+   * Input Source: `parameter.option`
+   * @deprecated this option is deprecated. Ignite sets you up to run native or Expo
+   * @default undefined
+   */
+  expo?: boolean
+  /**
+   * Package manager to install dependencies with
+   *
+   * Input Source: `prompt.ask`| `parameter.option`
+   */
+  packager?: "npm" | "yarn" | "pnpm"
+  /**
+   * The target directory where the project will be created.
+   *
+   * Input Source: `prompt.confirm` | `parameter.option`
+   * @default `${cwd}/${projectName}`
+   */
+  targetPath?: string
+  /**
+   * Whether or not to accept the default options for all prompts
+   *
+   * Input Source: `parameter.option`
+   * @default false
+   */
+  y?: boolean
+}
+
 export default {
   run: async (toolbox: GluegunToolbox) => {
-    const { print, filesystem, system, meta, parameters, strings } = toolbox
+    // #region Toolbox
+    const { print, filesystem, system, meta, parameters, strings, prompt } = toolbox
     const { kebabCase } = strings
     const { exists, path, remove, copy, read, write } = filesystem
     const { info, colors, warning } = print
     const { gray, red, magenta, cyan, yellow, green } = colors
+    const options: Options = parameters.options
+
+    const y = !!options.y
+    const useDefault = (option: unknown) => y && option === undefined
+    // #endregion
 
+    // #region Debug
     // start tracking performance
     const perfStart = new Date().getTime()
 
     // debug?
-    const debug = Boolean(parameters.options.debug)
+    const debug = Boolean(options.debug)
     const log = <T = unknown>(m: T): T => {
       debug && info(` ${m}`)
       return m
     }
 
     // log raw parameters for debugging
     log(`ignite command: ${parameters.argv.join(" ")}`)
+    // #endregion
+
+    // #region Project Name
+    heading("🔥 About to Ignite your new app! 🔥")
 
     // retrieve project name from toolbox
-    const { validateProjectName, validateBundleIdentifier } = require("../tools/validations")
-    const projectName = validateProjectName(toolbox)
+    const { validateProjectName } = require("../tools/validations") as ValidationsExports
+    const projectName = await validateProjectName(toolbox)
```

Comment:
Added a `prompt.ask` inside the validator if the project name is empty: https://github.com/infinitered/ignite/pull/1999/files#diff-7920917860492dfa617b4e6b53368ad95f26db81b3edb99a38336c578d56971bR12-R16